catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java

```diff
@@ -106,25 +107,45 @@ public int getMaxNumResults() {
         return maxNumResults;
     }
 
+    private void setLibraryQuerySources(Set<String> querySources) {
+        Optional.ofNullable(library)
+                .ifPresent(l -> l.setQuerySources(querySources));
+    }
+
     public void setMaxNumResults(int maxNumResults) {
         this.maxNumResults = maxNumResults;
-        if (library != null) {
-            library.setMaxNumResults(maxNumResults);
-        }
+        Optional.ofNullable(library)
+                .ifPresent(l -> l.setMaxNumResults(maxNumResults));
     }
 
-    public List<String> getQuerySources() {
+    @Override
+    public Set<String> getQuerySources() {
         return querySources;
     }
 
-    public void setQuerySources(List<String> querySources) {
-        this.querySources.clear();
+    public void setQuerySources(Set<String> querySources) {
+        //TODO? check if framework.getSourceIds().containsAll(querySources)
```

Comment:
I think it's ok how it is, but we would want to move the `this.querySources.clear()` below outside of the null check to ensure that if someone set the querySources to null we would clear the collection. Basically, we want `this.querySources` to only contain what is being set and nothing more. `.containsAll` would check if everything being set is in the collection, but not if there were additional things that the operator wished to have removed.